### PR TITLE
Add HTTP content and accept encoding helpers

### PR DESCRIPTION
**Pull request by tanner0101**
Adds helpers for working with `content-encoding` and `accept-encoding` headers (#2465). 

- Adds `HTTPHeaders.acceptEncoding`.

```swift
// Accept gzip and deflate encodings.
req.headers.acceptEncoding = [.gzip, .deflate]

// Check if request accepts gzip.
if req.headers.acceptEncoding.contains(.gzip) {
    // Request accepts gzip.    
}
```

- Adds `HTTPHeaders.contentEncoding`.

```swift
// Check if response is gzipped.
if res.headers.contentEncoding == .gzip {
    // Response is gzipped.
}
```

- Adds `HTTPStatus.class`.

```swift
// Check response's HTTP status class.
switch res.status {
case .informational: // 1xx
case .successful: // 2xx
case .redirection: // 3xx
case .clientError: // 4xx
case .serverError: // 5xx
default: // invalid
}
```

- Fixes a bug parsing some HTTP header values where commas would appear before semicolons.

### CODE REVIEW

**Comment by tanner0101** [comment]
Tests failing because of https://github.com/apple/swift-nio-extras/pull/105

File: Package.swift [hunk 1]
```diff
@@ -40,7 +40,7 @@ let package = Package(
         .package(url: "https://github.com/apple/swift-nio-http2.git", from: "1.13.0"),
         
         // Useful code around SwiftNIO.
-        .package(url: "https://github.com/apple/swift-nio-extras.git", from: "1.0.0"),
+        .package(url: "https://github.com/apple/swift-nio-extras.git", .branch("master")),
```
Comment (tanner0101) [inline Package.swift:43]: Cannot merge until this is fixed.

File: Sources/Vapor/HTTP/Headers/HTTPHeaders+Accept.swift [hunk 2]
```diff
@@ -0,0 +1,17 @@
+extension HTTPHeaders {
+    /// Returns a collection of `MediaTypePreference`s specified by this HTTP message's `"Accept"` header.
+    ///
+    /// You can returns all `MediaType`s in this collection to check membership.
```
Comment (siemensikkema) [inline Sources/Vapor/HTTP/Headers/HTTPHeaders+Accept.swift:4]: \* return

**Comment by 0xTim** [comment]
Closing due to time

**Pull request closed without merging**